crates/esthri-cli/src/http_server.rs

```diff
@@ -224,10 +231,12 @@ fn with_s3_client(
 pub fn esthri_filter(
     s3_client: S3Client,
     bucket: &str,
+    allowed_prefixes: &[String],
 ) -> impl Filter<Extract = (http::Response<Body>,), Error = warp::Rejection> + Clone {
     warp::path::full()
         .and(with_s3_client(s3_client))
         .and(with_bucket(bucket.to_owned()))
+        .and(with_allowed_prefixes(allowed_prefixes))
         .and(warp::query::<DownloadParams>())
         .and(warp::header::optional::<String>("if-none-match"))
         .and_then(download)
```

Comment:
Is it possible to do something like 

```rust
let path = warp::path::full()
    .and(with_s3_client(s3_client))        
    .and(with_bucket(bucket.to_owned()))
    .and(with_allowed_prefixes(allowed_prefixes))
    .and(warp::query::<DownloadParams>())
    .and(warp::query::<DownloadParams>())
    .and(warp::header::optional::<String>("if-none-match"))
    .and(warp::header::optional::<String>("if-none-match"))
    .and_then(download)
    .and_then(download)
if !allowed_prefixes.is_empty() {
    path.and(with_allowed_prefixes(allowed_prefixes));
} else {
    path
}
```
then the filter doesn't have to worry about an empty prefix list

Reply:
First thought it probably not?  This creates a function signature, so if we conditionally add it, then it would produce two diferent function signatures.

Reply:
I was able to do some preprocessing in the filter setup, which makes the code a little clearer